clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/PartitionReplicaIngestionContext.java

```diff
@@ -61,22 +58,14 @@ public WorkloadType getWorkloadType() {
     return workloadType;
   }
 
-  public static WorkloadType getWorkloadType(PubSubTopic versionTopic, Store store) {
-    checkVersionTopicStoreOrThrow(versionTopic, store);
-    int versionNumber = Version.parseVersionFromKafkaTopicName(versionTopic.getName());
-    Version version = store.getVersion(versionNumber);
-    if (version == null) {
-      return WorkloadType.NON_AA_OR_WRITE_COMPUTE;
-    }
-    if (store.isWriteComputationEnabled() || version.isActiveActiveReplicationEnabled()) {
+  public static WorkloadType getWorkloadType(boolean isActiveActiveReplicationEnabled, Store store) {
+    if (store.isWriteComputationEnabled() || isActiveActiveReplicationEnabled) {
       return WorkloadType.AA_OR_WRITE_COMPUTE;
     }
     return WorkloadType.NON_AA_OR_WRITE_COMPUTE;
   }
 
-  public static VersionRole getStoreVersionRole(PubSubTopic versionTopic, Store store) {
-    checkVersionTopicStoreOrThrow(versionTopic, store);
-    int versionNumber = Version.parseVersionFromKafkaTopicName(versionTopic.getName());
+  public static VersionRole getStoreVersionRole(int versionNumber, Store store) {
```

Comment:
What happens in case of version rollbacks? Backup and future version pools might be incorrect? 

Since there are returns in each conditions we don't need else statements right
Can we do
```
int currentVersionNumber = store.getCurrentVersion();

if (currentVersionNumber < versionNumber) {
    return VersionRole.FUTURE;
}
if (currentVersionNumber > versionNumber) {
    return VersionRole.BACKUP;
}
return VersionRole.CURRENT;
```